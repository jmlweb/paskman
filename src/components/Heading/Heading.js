import React from 'react';
import styled from 'styled-components';
import styledMap from 'styled-map';
import colors from '../../styles/colors';
import spacing from '../../styles/spacing';

const StyledHeadingMaster = styled.h1`
  color: ${props => colors[props.colorScheme]};
  font-size: ${styledMap('level', {
    '1': '2.2rem',
    '2': '2rem',
    '3': '1.8rem',
    '4': '1.6rem',
    '5': '1.4rem',
    '6': '1.2rem',
  })};
  font-weight: ${props => props.weight};
  margin: 0 0 ${spacing.md};
`;

const createStyledHeading = tag => StyledHeadingMaster.withComponent(tag);

const normalizeTag = (tag, level) => {
  if (tag) {
    return tag;
  }
  return `h${level}`;
};

const normalizeLevel = level => {
  if (level < 1) {
    return 1;
  }
  if (level > 6) {
    return 6;
  }
  return level;
};

const Heading = props => {
  const {level = '1', color = 'secondary', weight = 500, tag, children, ...rest} = props;
  const StyledHeading = createStyledHeading(normalizeTag(tag, normalizeLevel(+level)));
  return <StyledHeading colorScheme={color} level={level} weight={weight} {...rest}>{children}</StyledHeading>;
};

export default Heading;
