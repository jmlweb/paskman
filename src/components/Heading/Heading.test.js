import React from 'react';
import { configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import Heading from './Heading';

configure({ adapter: new Adapter() });

describe('Heading', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = mount(<Heading>Dummy</Heading>);
  });
  it('Render the component', () => {
    expect(wrapper.find(Heading).length).toEqual(1);
  });
  it('Render the component with tag', () => {
    wrapper.setProps({level: '2', color: 'primary', tag: 'p'});
    expect(wrapper.find(Heading).length).toEqual(1);
  });
  it('Render the component with level lower than 1', () => {
    wrapper.setProps({level: '0'});
    expect(wrapper.find(Heading).length).toEqual(1);
  });
  it('Render the component with level heigher than 6', () => {
    wrapper.setProps({level: '7'});
    expect(wrapper.find(Heading).length).toEqual(1);
  });
  it('Matchs snapshot', () => {
    const tree = renderer.create(wrapper).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
