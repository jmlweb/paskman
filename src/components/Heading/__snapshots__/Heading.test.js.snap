// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading Matchs snapshot 1`] = `
<h1
  className="sc-dnqmqq eahMDG"
>
  Dummy
</h1>
`;
